inbm/dispatcher-agent/dispatcher/fota/bios_factory.py

```diff
@@ -195,6 +194,20 @@ def __init__(self,
                  params: Dict) -> None:
         super().__init__(dispatcher_broker, repo, params)
 
+    def _get_guid_from_system(self, manifest_guid: Optional[str]) -> str:
+        # get the GUID from the system using FW tool
+        extracted_guids = extract_guids(
+            self._fw_tool, ["System Firmware type", "system-firmware type"])
+        if manifest_guid:
+            if manifest_guid not in extracted_guids:
+                raise FotaError(
```

Comment:
<h3>$\color{#FFD700}Coding Style$</h3>

Consider logging an informative message before raising `FotaError` when the GUID in the manifest does not match any system firmware GUID. This could aid in debugging.